src/meta-srv/src/metasrv/builder.rs

```diff
@@ -195,6 +196,9 @@ impl MetasrvBuilder {
         let flow_metadata_manager = Arc::new(FlowMetadataManager::new(
             leader_cached_kv_backend.clone() as _,
         ));
+        let maintenance_mode_manager = Arc::new(MaintenanceModeManager::new(
+            leader_cached_kv_backend.clone() as _,
```

Comment:
Seems we cannot use `leader_cached_kv_backend` since a follower can also set `maintenance_mode`